gitwatch.go

```diff
@@ -235,3 +251,30 @@ func GetRepoPath(cache, repo string) (result string, err error) {
 	}
 	return filepath.Join(cache, filepath.Base(u.Path)), nil
 }
+
+// MakeRepositoryList Creates a repository list from an array of
+// strings, while also checking is the string contains a special
+// character which can be used to get the branch to use
+func MakeRepositoryList(repos []string) (result []Repository) {
```

Comment:
I'd like to see a basic unit test for this, just to protect against future regressions and make sure it works as expected before merging to master (and, after, deploy to my live servers).

Reply:
Not sure how I would test this? Since if it didn't work, the rest of the tests would fail anyway.

Reply:
Just a basic Input -> Expected output test with a few cases. You can generate the unit test code in vscode with the "Go: Generate Unit Tests For Function" tool.